File: C/Hello_C/step_1/diff/diff.c
#include <stdio.h>
#include <stdlib.h>

int square(float x) {
    return x * x;
};

float diff(int x, float dx, int (f)(float)) {
    return (f(x + dx) - f(x))/dx;
}

float main() {
    float result = diff(10, 1e-10, square);
    sprintf(result);
    return result;
}

File: JS/PURE_JS/MATH/Calculus/diff.mjs
const square = x => x * x;

const dx1 = 1e-10;
const dx2 = 1e-15;

const diff = (x, dx = dx1, f = square) => (f(x + dx) - f(x)) / dx; 

export { diff, dx1, dx2, square };
